diff --git a/browser/components/BrowserGlue.sys.mjs b/browser/components/BrowserGlue.sys.mjs
index 67b3618..c3e646d 100644
--- a/browser/components/BrowserGlue.sys.mjs
+++ b/browser/components/BrowserGlue.sys.mjs
@@ -1501,7 +1501,6 @@ BrowserGlue.prototype = {
       lazy.Normandy.init();
     }
 
-    lazy.SaveToPocket.init();
 
     lazy.ResetPBMPanel.init();
 
diff --git a/browser/components/moz.build b/browser/components/moz.build
index 6cbb7ce..37d98cd 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -49,7 +49,6 @@ DIRS += [
     "originattributes",
     "pagedata",
     "places",
-    "pocket",
     "preferences",
     "privatebrowsing",
     "prompts",
diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 3b2b6e6..c46d84f 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -3630,7 +3630,6 @@ var XULBrowserWindow = {
 
     SafeBrowsingNotificationBox.onLocationChange(aLocationURI);
 
-    SaveToPocket.onLocationChange(window);
 
     let originalURI;
     if (aRequest instanceof Ci.nsIChannel) {
