diff --git a/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def b/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
index 5f6af35..07cbb46 100644
--- a/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
+++ b/src/llvm-project/llvm/include/llvm/Analysis/TargetLibraryInfo.def
@@ -41,6 +41,15 @@
 #define TLI_DEFINE_SIG_INTERNAL(...) { __VA_ARGS__ },
 #endif
 
+// avoid name conflicts with musl-libc
+#undef fopen64
+#undef fseeko64
+#undef ftello64
+#undef fstat64
+#undef lstat64
+#undef stat64
+#undef tmpfile64
+
 /// void *operator new(unsigned int);
 TLI_DEFINE_ENUM_INTERNAL(msvc_new_int)
 TLI_DEFINE_STRING_INTERNAL("??2@YAPAXI@Z")
