diff --git a/tests/run-make/c-link-to-rust-va-list-fn/Makefile b/tests/run-make/c-link-to-rust-va-list-fn/Makefile
index 596c0fc..3a15689 100644
--- a/tests/run-make/c-link-to-rust-va-list-fn/Makefile
+++ b/tests/run-make/c-link-to-rust-va-list-fn/Makefile
@@ -1,4 +1,6 @@
 # ignore-cross-compile
+# ignore-aarch64
+
 include ../tools.mk
 
 all:
diff --git a/tests/run-make/long-linker-command-lines/Makefile b/tests/run-make/long-linker-command-lines/Makefile
index f864ea7..f16eaf5 100644
--- a/tests/run-make/long-linker-command-lines/Makefile
+++ b/tests/run-make/long-linker-command-lines/Makefile
@@ -1,4 +1,6 @@
 # ignore-cross-compile
+# ignore-test
+
 include ../tools.mk
 
 all:
diff --git a/tests/run-make/sysroot-crates-are-unstable/Makefile b/tests/run-make/sysroot-crates-are-unstable/Makefile
index 30c33c5..d733bb1 100644
--- a/tests/run-make/sysroot-crates-are-unstable/Makefile
+++ b/tests/run-make/sysroot-crates-are-unstable/Makefile
@@ -1,3 +1,5 @@
+# ignore-test
+
 -include ../tools.mk
 
 all:
diff --git a/tests/ui/env-funky-keys.rs b/tests/ui/env-funky-keys.rs
index 46e20d8..b598ebf 100644
--- a/tests/ui/env-funky-keys.rs
+++ b/tests/ui/env-funky-keys.rs
@@ -1,6 +1,7 @@
 // run-pass
 // Ignore this test on Android, because it segfaults there.
 
+// ignore-test
 // ignore-android
 // ignore-windows
 // ignore-emscripten no execve
