--- a/gcc/tree-vect-generic.c	2018/11/26 11:57:16	266457
+++ b/gcc/tree-vect-generic.c	2018/11/26 11:59:05	266458
@@ -89,12 +89,8 @@
       return vector_last_type;
     }
 
-  /* We build a new type, but we canonicalize it nevertheless,
-     because it still saves some memory.  */
   vector_last_nunits = nunits;
-  vector_last_type = type_hash_canon (nunits,
-				      build_vector_type (vector_inner_type,
-							 nunits));
+  vector_last_type = build_vector_type (vector_inner_type, nunits);
   return vector_last_type;
 }
 

