--- branches/gcc-7-branch/gcc/ipa-prop.c	2019/01/16 15:37:33	267974
+++ branches/gcc-7-branch/gcc/ipa-prop.c	2019/01/16 15:41:07	267975
@@ -1577,7 +1577,8 @@
       if (TREE_CODE (arg) == SSA_NAME)
 	{
 	  tree type_size;
-          if (!tree_fits_uhwi_p (TYPE_SIZE (TREE_TYPE (arg_type))))
+          if (!tree_fits_uhwi_p (TYPE_SIZE (TREE_TYPE (arg_type)))
+	      || !POINTER_TYPE_P (TREE_TYPE (arg)))
             return;
 	  check_ref = true;
 	  arg_base = arg;
