From 9923a5d7b16cedafcf23c3d6494f0c1c2df906ab Mon Sep 17 00:00:00 2001
From: Alexey Yakovenko <wakeroid@gmail.com>
Date: Tue, 4 Feb 2020 21:47:03 +0100
Subject: [PATCH] alsa: pause/unpause device when setting format (fixes #2192)

---
 plugins/alsa/alsa.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/plugins/alsa/alsa.c b/plugins/alsa/alsa.c
index 020b7e4a8..87da0f1cc 100644
--- a/plugins/alsa/alsa.c
+++ b/plugins/alsa/alsa.c
@@ -71,6 +71,9 @@ palsa_free (void);
 static int
 palsa_setformat (ddb_waveformat_t *fmt);
 
+static ddb_playback_state_t
+palsa_get_state (void);
+
 static int
 palsa_play (void);
 
@@ -424,6 +427,10 @@ palsa_init (void) {
 static int
 palsa_setformat (ddb_waveformat_t *fmt) {
     LOCK;
+    ddb_playback_state_t state = palsa_get_state ();
+    if (state == DDB_PLAYBACK_STATE_PLAYING) {
+        palsa_pause ();
+    }
     memcpy (&requested_fmt, fmt, sizeof (ddb_waveformat_t));
     trace ("palsa_setformat %dbit %s %dch %dHz channelmask=%X\n", requested_fmt.bps, fmt->is_float ? "float" : "int", fmt->channels, fmt->samplerate, fmt->channelmask);
     if (!audio
@@ -454,6 +461,9 @@ palsa_setformat (ddb_waveformat_t *fmt) {
         return -1;
     }
     trace ("new format %dbit %s %dch %dHz channelmask=%X\n", plugin.fmt.bps, plugin.fmt.is_float ? "float" : "int", plugin.fmt.channels, plugin.fmt.samplerate, plugin.fmt.channelmask);
+    if (state == DDB_PLAYBACK_STATE_PLAYING) {
+        palsa_unpause ();
+    }
     UNLOCK;
 
     return 0;
