From 49c1f8ae21572a35fb5acd6e7f5f8d5d378c70eb Mon Sep 17 00:00:00 2001
From: Alexey Yakovenko <wakeroid@gmail.com>
Date: Sun, 2 Feb 2020 10:30:05 +0100
Subject: [PATCH] streamer: move streamer_reset to after output->stop (fixes
 #2192)

---
 streamer.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/streamer.c
+++ b/streamer.c
@@ -2255,10 +2255,9 @@ play_index (int idx, int startpaused) {
 
 error:
     output->stop ();
-
-    streamer_lock();
     streamer_reset (1);
 
+    streamer_lock();
     _handle_playback_stopped ();
     stream_track (NULL, 0);
     if (plt) {
@@ -2315,7 +2314,6 @@ play_current (void) {
 static void
 play_next (int dir) {
     DB_output_t *output = plug_get_output ();
-    streamer_reset(1);
 
     playItem_t *origin = NULL;
     if (buffering_track) {
@@ -2360,6 +2358,7 @@ play_next (int dir) {
 
     if (!next) {
         output->stop ();
+        streamer_reset(1);
         _handle_playback_stopped ();
         return;
     }
