--- a/m4/ul.m4	2018-04-24 22:22:47.322071477 -0600
+++ b/m4/ul.m4	2018-04-24 22:23:10.082408833 -0600
@@ -461,8 +461,8 @@
     AC_CHECK_TOOL(SUFFIX[]6_CONFIG, suffix[]6-config)
     if AC_RUN_LOG([$SUFFIX[]6_CONFIG --version >/dev/null]); then
       have_[]suffix=yes
-      NCURSES_LIBS=`$SUFFIX[]6_CONFIG --libs`
-      NCURSES_CFLAGS=`$SUFFIX[]6_CONFIG --cflags`
+      NCURSES_LIBS=`/usr/armv7a-unknown-linux-gnueabihf/usr/bin/ncursesw6-config --libs`
+      NCURSES_CFLAGS=`/usr/armv7a-unknown-linux-gnueabihf/usr/bin/ncursesw6-config --cflags`
     else
       have_[]suffix=no
     fi

