diff -upr wine-1.7.39/dlls/wined3d/context.c wine-1.7.39-r1/dlls/wined3d/context.c
--- wine-1.7.39/dlls/wined3d/context.c	2015-03-20 14:41:44.000000000 +0100
+++ wine-1.7.39-r1/dlls/wined3d/context.c	2015-03-22 12:29:49.187038020 +0100
@@ -1463,10 +1463,14 @@ struct wined3d_context *context_create(s
         }
     }
 
-    if (!(hdc = GetDC(swapchain->win_handle)))
-    {
+    hdc = NULL;
+
+    if (swapchain->win_handle == GetDesktopWindow())
+        TRACE("Attempted to retrieve device context of desktop window, trying swapchain backup.\n");
+    else if (!(hdc = GetDC(swapchain->win_handle)))
         WARN("Failed to retrieve device context, trying swapchain backup.\n");
-
+    if (!hdc)
+    {
         if ((hdc = swapchain_get_backup_dc(swapchain)))
             hdc_is_private = TRUE;
         else
