--- a/gcc/tree-outof-ssa.c	2019/08/29 14:06:32	275034
+++ b/gcc/tree-outof-ssa.c	2019/08/29 14:57:18	275035
@@ -651,6 +651,8 @@
   tree type = TREE_TYPE (name);
   int unsignedp;
   machine_mode reg_mode = promote_ssa_mode (name, &unsignedp);
+  if (reg_mode == BLKmode)
+    return assign_temp (type, 0, 0);
   rtx x = gen_reg_rtx (reg_mode);
   if (POINTER_TYPE_P (type))
     mark_reg_pointer (x, TYPE_ALIGN (TREE_TYPE (type)));

