diff --git a/utils/getconf.c b/utils/getconf.c
index 3dd3d75..f0bfb24 100644
--- a/utils/getconf.c
+++ b/utils/getconf.c
@@ -1062,9 +1062,9 @@ usage (void)
 {
   fprintf (stderr,
 	   _("Usage: %s [-v specification] variable_name [pathname]\n"),
-	   __progname);
+	   ptsname);
   fprintf (stderr,
-	   _("       %s -a [pathname]\n"), __progname);
+	   _("       %s -a [pathname]\n"), ptsname);
   exit (2);
 }
 
