diff --git a/CMakeLists.txt b/CMakeLists.txt
index 590734d..a1fb6b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,8 +80,11 @@ ELSE()
 		SET(PLATFORM_ARCH "x86_64")
 	ELSEIF(("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "universal" OR "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "") AND APPLE)
 		SET(PLATFORM_ARCH "universal")
+#	ELSEIF("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "^armv7*"
+#		SET(PLATFORM_ARCH "arm")
 	ELSE()
-		MESSAGE(FATAL_ERROR "Unknown processor arch: ${CMAKE_SYSTEM_PROCESSOR}")
+#		MESSAGE(FATAL_ERROR "Unknown processor arch: ${CMAKE_SYSTEM_PROCESSOR}")
+		SET(PLATFORM_ARCH "arm")
 	ENDIF()
 ENDIF()
 # Fill cache for post-processing tools
diff --git a/cmake/linux.cmake b/cmake/linux.cmake
index cb95d44..3a9dd6e 100644
--- a/cmake/linux.cmake
+++ b/cmake/linux.cmake
@@ -11,6 +11,6 @@ endif(CCACHE_FOUND)
 
 # set compiler flags
 IF(RTTR_ENABLE_OPTIMIZATIONS)
-	FORCE_ADD_FLAGS(CMAKE_C_FLAGS -ffast-math -mmmx -msse -mfpmath=sse -ggdb)
-	FORCE_ADD_FLAGS(CMAKE_CXX_FLAGS -ffast-math -mmmx -msse -mfpmath=sse -ggdb)
+	FORCE_ADD_FLAGS(CMAKE_C_FLAGS -Ofast -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard)
+	FORCE_ADD_FLAGS(CMAKE_CXX_FLAGS -Ofast -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard)
 ENDIF(RTTR_ENABLE_OPTIMIZATIONS)
