From c0ebcfd3d6647b0cc77de442582a06011a7f8cad Mon Sep 17 00:00:00 2001
From: Flamefire <Flamefire@users.noreply.github.com>
Date: Thu, 2 Jan 2020 16:08:49 +0100
Subject: [PATCH] Fix to narrow tooltips

Fixes #1154
---
 libs/s25main/WindowManager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libs/s25main/WindowManager.cpp b/libs/s25main/WindowManager.cpp
index 74782c18e..0ab345bde 100644
--- a/libs/s25main/WindowManager.cpp
+++ b/libs/s25main/WindowManager.cpp
@@ -796,9 +796,9 @@ class WindowManager::Tooltip
     {
         if(lines.empty())
             return;
-        width = std::numeric_limits<unsigned>::max();
+        width = 0;
         for(const auto& line : lines)
-            width = std::min(width, font->getWidth(line));
+            width = std::max(width, font->getWidth(line));
         width += BORDER_SIZE * 2;
         height = lines.size() * font->getHeight() + BORDER_SIZE * 2;
     }
