diff --git a/src/Debug.cpp b/src/Debug.cpp
index 0af1e4b..d1ed3e7 100644
--- a/src/Debug.cpp
+++ b/src/Debug.cpp
@@ -224,7 +224,7 @@ bool DebugInfo::SendStackTrace()
 	LOG.lprintf("Read Frames %d\n", num_frames);
 	*/
 #else
-	unsigned num_frames = backtrace(stacktrace, maxTrace);
+	unsigned num_frames = 0;
 #endif
 
 	LOG.lprintf("Will now send %d stack frames\n", num_frames);
diff --git a/src/Debug.h b/src/Debug.h
index 9a31264..2808c2b 100644
--- a/src/Debug.h
+++ b/src/Debug.h
@@ -27,8 +27,6 @@
 #ifdef _WIN32
 #	include <windows.h>
 #	include <dbghelp.h>
-#else
-#	include <execinfo.h>
 #endif
 
 class DebugInfo : Socket

