diff --git a/libutil/src/Socket.cpp b/libutil/src/Socket.cpp
index 633123b..7b4e463 100644
--- a/libutil/src/Socket.cpp
+++ b/libutil/src/Socket.cpp
@@ -622,7 +622,7 @@ bool Socket::Connect(const std::string& hostname, const unsigned short port, boo
                 Sleep(50);
 
                 ++timeout;
-                if(timeout > 8)
+                if(timeout > 800)
                 {
 #ifdef _WIN32
                     WSASetLastError(WSAETIMEDOUT);
