From a213aedbdced13cc7c4080ac7370e532c42de136 Mon Sep 17 00:00:00 2001
From: Olivier Duchateau <duchateau.olivier@gmail.com>
Date: Sun, 28 Jun 2020 21:29:21 +0200
Subject: [PATCH] Fix GTimeVal deprecation

---
 panel-plugin/weather-summary.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/panel-plugin/weather-summary.c b/panel-plugin/weather-summary.c
index f5bfdf67..2ceef322 100644
--- a/panel-plugin/weather-summary.c
+++ b/panel-plugin/weather-summary.c
@@ -1124,7 +1124,6 @@ gboolean
 update_summary_subtitle(plugin_data *data)
 {
     time_t now_t;
-    GTimeVal now;
     gchar *title, *date;
     guint update_interval;
     gint64 now_ms;
@@ -1147,8 +1146,7 @@ update_summary_subtitle(plugin_data *data)
     g_free(title);
 
     /* compute and schedule the next update */
-    g_get_current_time(&now);
-    now_ms = ((gint64) now.tv_sec * 1000) + ((gint64) now.tv_usec / 1000);
+    now_ms = g_get_real_time () / 1000;
 #ifdef HAVE_UPOWER_GLIB
     if (data->upower_on_battery)
         update_interval = 60000 - (now_ms % 60000) + 10;
-- 
GitLab


