diff --git a/isocorespin b/isocorespin
index 7798597..af0e411 100644
--- a/isocorespin
+++ b/isocorespin
@@ -766,7 +766,7 @@ function CHECK_PACKAGE_DEPENDENCIES {
 	[ ! $(sudo bash -c "command -v losetup") ] && echo "${0}: Please ensure package 'klibc-utils' or equivalent for your distro is installed." && exit
 	[ ! $(sudo bash -c "command -v ip") ] && echo "${0}: Please ensure package 'iproute2' or equivalent for your distro is installed." && exit
 	[ ! $(sudo bash -c "command -v isoinfo") ] && echo "${0}: Please ensure package 'genisoimage' or equivalent for your distro is installed." && exit
-	[ ! $(sudo bash -c "command -v mkdosfs") ] && echo "${0}: Please ensure package 'dosfstools' or equivalent for your distro is installed." && exit
+	[ ! $(sudo bash -c "command -v mkfs.fat") ] && echo "${0}: Please ensure package 'dosfstools' or equivalent for your distro is installed." && exit
 	[ ! $(sudo bash -c "command -v mksquashfs") ] && echo "${0}: Please ensure package 'squashfs-tools' or equivalent for your distro is installed." && exit
 	[ ! $(sudo bash -c "command -v rsync") ] && echo "${0}: Please ensure package 'rsync' or equivalent for your distro is installed." && exit
 	[ ! $(sudo bash -c "command -v unsquashfs") ] && echo "${0}: Please ensure package 'squashfs-tools' or equivalent for your distro is installed." && exit
@@ -3031,7 +3031,7 @@ menuentry \"LINUXIUM ISO\" {\\
 		sudo umount mnt
 		sudo rm -f iso-directory-structure/boot/grub/efi.img
 		sudo dd if=/dev/zero of=iso-directory-structure/boot/grub/efi.img bs=1 count=3109888 > /dev/null 2>&1
-		sudo mkdosfs iso-directory-structure/boot/grub/efi.img > /dev/null 2>&1
+		sudo mkfs.fat iso-directory-structure/boot/grub/efi.img > /dev/null 2>&1
 		sudo mount iso-directory-structure/boot/grub/efi.img mnt
 		sudo cp -a efi mnt
 		sudo cp grub_bootia32.efi mnt/efi/boot/bootia32.efi
