From 62802fee96f002134719d02158c6fe859847fd9b Mon Sep 17 00:00:00 2001
From: Peter Dimov <pdimov@pdimov.com>
Date: Sat, 23 Dec 2017 17:22:35 +0200
Subject: [PATCH] Add unaligned test

---
 libs/endian/test/buffer_test.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/test/buffer_test.cpp b/test/buffer_test.cpp
index e214a03..cb305f2 100644
--- a/libs/endian/test/buffer_test.cpp
+++ b/libs/endian/test/buffer_test.cpp
@@ -145,6 +145,12 @@ namespace
 
   }
 
+  template<class T> struct unaligned
+  {
+    char x;
+    T y;
+  };
+
   template<class T> void test_buffer_type( typename T::value_type v1, typename T::value_type v2 )
   {
     T buffer( v1 );
@@ -152,6 +158,12 @@ namespace
 
     buffer = v2;
     BOOST_TEST_EQ( buffer.value(), v2 );
+
+    unaligned<T> buffer2 = { 0, T( v1 ) };
+    BOOST_TEST_EQ( buffer2.y.value(), v1 );
+
+    buffer2.y = v2;
+    BOOST_TEST_EQ( buffer2.y.value(), v2 );
   }
 
   void test_construction_and_assignment()
