--- a/meson.build	2026-01-05 09:40:06.658542988 +0100
+++ b/meson.build	2026-01-05 09:40:10.910174438 +0100
@@ -3310,26 +3310,19 @@
 #                      install_dir : xinitrcdir)
 # endif
 #endif // 0
-install_data('LICENSE.GPL2',
-             'LICENSE.LGPL2.1',
 #if 0 /// elogind has upgraded to markdown, and the NEWS file is useless
 #              'NEWS',
 #              'README',
 #else // 0
-             'README.md',
 #endif // 0
-             'docs/CODING_STYLE.md',
 #if 0 /// irrelevant for elogind
 #              'docs/DISTRO_PORTING.md',
 #endif // 0
-             'docs/ENVIRONMENT.md',
 #if 0 /// irrelevant for elogind
 #              'docs/HACKING.md',
 #              'docs/TRANSIENT-SETTINGS.md',
 #              'docs/TRANSLATORS.md',
 #endif // 0
-             'docs/UIDS-GIDS.md',
-             install_dir : docdir)
 
 #if 0 /// irrelevant for elogind
 # install_subdir('LICENSES',
