diff --git a/browser/extensions/moz.build b/browser/extensions/moz.build
index 3c6e7eb..2df11e8 100644
--- a/browser/extensions/moz.build
+++ b/browser/extensions/moz.build
@@ -5,10 +5,4 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DIRS += [
-    "formautofill",
-    "screenshots",
-    "webcompat",
-    "report-site-issue",
-    "pictureinpicture",
-    "search-detection",
 ]
diff --git a/browser/locales/Makefile.in b/browser/locales/Makefile.in
index 1b8f21b..8f0fbf6 100644
--- a/browser/locales/Makefile.in
+++ b/browser/locales/Makefile.in
@@ -55,7 +55,6 @@ l10n-%:
 ifneq (,$(wildcard ../extensions/formautofill/locales))
 	@$(MAKE) -C ../extensions/formautofill/locales AB_CD=$* XPI_NAME=locale-$*
 endif
-	@$(MAKE) -C ../extensions/report-site-issue/locales AB_CD=$* XPI_NAME=locale-$*
 	@$(MAKE) -C ../../devtools/client/locales AB_CD=$* XPI_NAME=locale-$* XPI_ROOT_APPID='$(XPI_ROOT_APPID)'
 	@$(MAKE) -C ../../devtools/startup/locales AB_CD=$* XPI_NAME=locale-$* XPI_ROOT_APPID='$(XPI_ROOT_APPID)'
 	@$(MAKE) l10n AB_CD=$* XPI_NAME=locale-$* PREF_DIR=$(PREF_DIR)
@@ -76,7 +75,6 @@ endif
 	@$(MAKE) -C ../../devtools/startup/locales chrome AB_CD=$*
 	@$(MAKE) chrome AB_CD=$*
 	@$(MAKE) -C $(DEPTH)/$(MOZ_BRANDING_DIRECTORY)/locales chrome AB_CD=$*
-	@$(MAKE) -C ../extensions/report-site-issue/locales chrome AB_CD=$*
 
 package-win32-installer: $(SUBMAKEFILES)
 	$(MAKE) -C ../installer/windows CONFIG_DIR=l10ngen ZIP_IN='$(ZIP_OUT)' installer
diff --git a/browser/locales/filter.py b/browser/locales/filter.py
index 9155947..e33d121 100644
--- a/browser/locales/filter.py
+++ b/browser/locales/filter.py
@@ -14,8 +14,6 @@ def test(mod, path, entity=None):
         "devtools/shared",
         "devtools/startup",
         "browser",
-        "browser/extensions/formautofill",
-        "browser/extensions/report-site-issue",
         "extensions/spellcheck",
         "other-licenses/branding/firefox",
         "browser/branding/official",
diff --git a/browser/locales/l10n.ini b/browser/locales/l10n.ini
index fad1533..76fdaff 100644
--- a/browser/locales/l10n.ini
+++ b/browser/locales/l10n.ini
@@ -12,8 +12,6 @@ dirs = browser
      browser/branding/official
      devtools/client
      devtools/startup
-     browser/extensions/formautofill
-     browser/extensions/report-site-issue
 
 [includes]
 # non-central apps might want to use %(topsrcdir)s here, or other vars
diff --git a/browser/locales/l10n.toml b/browser/locales/l10n.toml
index 31baf8c..baaff3b 100644
--- a/browser/locales/l10n.toml
+++ b/browser/locales/l10n.toml
@@ -131,15 +131,6 @@ locales = [
     reference = "browser/branding/official/locales/en-US/**"
     l10n = "{l}browser/branding/official/**"
 
-[[paths]]
-    reference = "browser/extensions/formautofill/locales/en-US/**"
-    l10n = "{l}browser/extensions/formautofill/**"
-
-[[paths]]
-    reference = "browser/extensions/report-site-issue/locales/en-US/**"
-    l10n = "{l}browser/extensions/report-site-issue/**"
-
-
 [[includes]]
     path = "toolkit/locales/l10n.toml"
 
