diff --git a/browser/components/newtab/data/content/activity-stream.bundle.js b/browser/components/newtab/data/content/activity-stream.bundle.js
index e064ee4..25d8d81 100644
--- a/browser/components/newtab/data/content/activity-stream.bundle.js
+++ b/browser/components/newtab/data/content/activity-stream.bundle.js
@@ -1783,7 +1783,7 @@ class ASRouterAdminInner extends (external_React_default()).PureComponent {
         label = /*#__PURE__*/external_React_default().createElement("span", null, "remote settings (", /*#__PURE__*/external_React_default().createElement("a", {
           className: "providerUrl",
           target: "_blank",
-          href: "https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/nimbus-desktop-experiments/records",
+          href: "https://see.ticket.about.this.invalid",
           rel: "noopener noreferrer"
         }, "nimbus-desktop-experiments"), ")");
       }
@@ -15287,4 +15287,4 @@ function renderCache(initialState) {
 }
 NewtabRenderUtils = __webpack_exports__;
 /******/ })()
-;
\ No newline at end of file
+;
diff --git a/services/settings/Utils.sys.mjs b/services/settings/Utils.sys.mjs
index 7d16f40..aeb8e10 100644
--- a/services/settings/Utils.sys.mjs
+++ b/services/settings/Utils.sys.mjs
@@ -51,6 +51,9 @@ XPCOMUtils.defineLazyGetter(lazy, "isRunningTests", () => {
 // Overriding the server URL is normally disabled on Beta and Release channels,
 // except under some conditions.
 XPCOMUtils.defineLazyGetter(lazy, "allowServerURLOverride", () => {
+
+    return true; // always allow this override for LW
+
   if (!AppConstants.RELEASE_OR_BETA) {
     // Always allow to override the server URL on Nightly/DevEdition.
     return true;
diff --git a/toolkit/components/search/SearchUtils.sys.mjs b/toolkit/components/search/SearchUtils.sys.mjs
index 53fb84c..c61b55f 100644
--- a/toolkit/components/search/SearchUtils.sys.mjs
+++ b/toolkit/components/search/SearchUtils.sys.mjs
@@ -152,9 +152,9 @@ export var SearchUtils = {
 
   ENGINES_URLS: {
     "prod-main":
-      "https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/search-config/records",
+      "https://%.invalid",
     "prod-preview":
-      "https://firefox.settings.services.mozilla.com/v1/buckets/main-preview/collections/search-config/records",
+      "https://%.invalid"
     "stage-main":
       "https://firefox.settings.services.allizom.org/v1/buckets/main/collections/search-config/records",
     "stage-preview":
