diff --git a/xpcom/threads/MozPromise.h b/xpcom/threads/MozPromise.h
index b9a4eba..581a130 100644
--- a/xpcom/threads/MozPromise.h
+++ b/xpcom/threads/MozPromise.h
@@ -672,7 +672,7 @@ class MozPromise : public MozPromiseBase {
   static void MaybeChain(PromiseType* aFrom,
                          RefPtr<typename PromiseType::Private>&& aTo) {
     if (aTo) {
-      MOZ_DIAGNOSTIC_ASSERT(
+      MOZ_RELEASE_ASSERT(
           aFrom,
           "Can't do promise chaining for a non-promise-returning method.");
       aFrom->ChainTo(aTo.forget(), "<chained completion promise>");
