diff --git a/build/moz.configure/compile-checks.configure b/build/moz.configure/compile-checks.configure
--- a/build/moz.configure/compile-checks.configure
+++ b/build/moz.configure/compile-checks.configure
@@ -133,7 +133,7 @@
         @depends(check_symbol_flags, dependable(flags))
         def flags(base_flags, extra_flags):
             if base_flags and extra_flags:
-                return base_flags + extra_flags
+                return base_flags + list(extra_flags)
             if extra_flags:
                 return extra_flags
             return base_flags


