diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -958,7 +958,7 @@
             path = sysroot_input[0]
         if path:
             log.info("Using %s sysroot in %s", host_or_target_str, path)
-        return namespace(path=path, bootstrapped=bool(sysroot_input))
+        return namespace(path=path, bootstrapped=bool(path and not sysroot_input))
 
     return sysroot_path
 


