diff --git a/dom/media/webaudio/AudioNodeEngine.cpp b/dom/media/webaudio/AudioNodeEngine.cpp
--- a/dom/media/webaudio/AudioNodeEngine.cpp
+++ b/dom/media/webaudio/AudioNodeEngine.cpp
@@ -7,19 +7,19 @@
 #include "AudioNodeEngine.h"
 
 #include "mozilla/AbstractThread.h"
 #ifdef USE_NEON
 #  include "mozilla/arm.h"
-#  include "AudioNodeEngineGeneric.h"
+#  include "AudioNodeEngineNEON.h"
 #endif
 #ifdef USE_SSE2
 #  include "mozilla/SSE.h"
-#  include "AudioNodeEngineGeneric.h"
+#  include "AudioNodeEngineSSE2.h"
 #endif
 #if defined(USE_SSE42) && defined(USE_FMA3)
 #  include "mozilla/SSE.h"
-#  include "AudioNodeEngineGeneric.h"
+#  include "AudioNodeEngineSSE4_2_FMA3.h"
 #endif
 #include "AudioBlock.h"
 #include "Tracing.h"
 
 namespace mozilla {
diff --git a/dom/media/webaudio/AudioNodeEngineNEON.h b/dom/media/webaudio/AudioNodeEngineNEON.h
new file mode 100644
--- /dev/null
+++ b/dom/media/webaudio/AudioNodeEngineNEON.h
@@ -0,0 +1,9 @@
+/* -*- mode: c++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* this source code form is subject to the terms of the mozilla public
+ * license, v. 2.0. if a copy of the mpl was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "AudioNodeEngineGeneric.h"
+namespace mozilla {
+extern template struct Engine<xsimd::neon>;
+}  // namespace mozilla
diff --git a/dom/media/webaudio/AudioNodeEngineSSE2.h b/dom/media/webaudio/AudioNodeEngineSSE2.h
new file mode 100644
--- /dev/null
+++ b/dom/media/webaudio/AudioNodeEngineSSE2.h
@@ -0,0 +1,10 @@
+/* -*- mode: c++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* this source code form is subject to the terms of the mozilla public
+ * license, v. 2.0. if a copy of the mpl was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "AudioNodeEngineGeneric.h"
+
+namespace mozilla {
+extern template struct Engine<xsimd::sse2>;
+}  // namespace mozilla
diff --git a/dom/media/webaudio/AudioNodeEngineSSE4_2_FMA3.h b/dom/media/webaudio/AudioNodeEngineSSE4_2_FMA3.h
new file mode 100644
--- /dev/null
+++ b/dom/media/webaudio/AudioNodeEngineSSE4_2_FMA3.h
@@ -0,0 +1,10 @@
+/* -*- mode: c++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* this source code form is subject to the terms of the mozilla public
+ * license, v. 2.0. if a copy of the mpl was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "AudioNodeEngineGeneric.h"
+
+namespace mozilla {
+extern template struct Engine<xsimd::fma3<xsimd::sse4_2>>;
+}  // namespace mozilla


