diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -918,13 +918,15 @@
 
 
 @template
-def sysroot_path(host_or_target):
+def sysroot_path(host_or_target, target_sysroot=None):
+    assert target_sysroot or host_or_target is target
     if host_or_target is host:
         host_or_target_str = "host"
         opt = "--with-host-sysroot"
         env = "HOST_SYSROOT"
         when = depends(host)(lambda h: h.kernel == "Linux")
-        bootstrap_when = when
+        # Only bootstrap a host sysroot when using a bootstrapped target sysroot
+        bootstrap_when = when & target_sysroot.bootstrapped
     else:
         assert host_or_target is target
         host_or_target_str = "target"
@@ -943,7 +945,8 @@
 
     sysroot_input = depends(opt, when=when)(lambda x: x)
     bootstrap_sysroot = depends(bootstrap_when, sysroot_input)(
-        lambda bootstrap, input: bootstrap and not input
+        # Only bootstrap when no flag was explicitly given (either --with or --without)
+        lambda bootstrap, input: bootstrap and not input and input.origin == "default"
     )
 
     @depends(
@@ -983,7 +986,7 @@
             )
 
 
-host_sysroot_path = sysroot_path(host).path
+host_sysroot_path = sysroot_path(host, target_sysroot_path).path
 target_sysroot_path = target_sysroot_path.path
 
 


