diff --git a/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c b/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
index 057b5d7..4a1779e 100644
--- a/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
+++ b/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
@@ -64,6 +64,7 @@ SQLITE_EXTENSION_INIT3
 #ifndef _WIN32
 #ifndef __EMSCRIPTEN__
 #ifndef __COSMOPOLITAN__
+#ifndef linux
 #ifndef __wasi__
 typedef u_int8_t uint8_t;
 typedef u_int16_t uint16_t;
@@ -72,6 +73,7 @@ typedef u_int64_t uint64_t;
 #endif
 #endif
 #endif
+#endif
 
 typedef int8_t i8;
 typedef uint8_t u8;
