diff --git a/build/moz.configure/flags.configure b/build/moz.configure/flags.configure
index 73a6253..cb4e05c 100644
--- a/build/moz.configure/flags.configure
+++ b/build/moz.configure/flags.configure
@@ -603,13 +603,6 @@ def host_ldflags(env_ldflags, linker_flags, host_linker_ldflags, host, compiler)
     return flags
 
 
-@depends("CPPFLAGS")
-@imports(_from="mozbuild.shellutil", _import="split")
-def os_cppflags(env_cppflags):
-    flags = split(env_cppflags[0])
-    return flags
-
-
 @depends("ASFLAGS", asm_flags, android_flags, all_arm_flags)
 @imports(_from="mozbuild.shellutil", _import="split")
 def os_asflags(env_asflags, asm_flags, android_flags, all_arm_flags):
@@ -630,8 +623,6 @@ add_old_configure_assignment(
     "_COMPILATION_OPTIMIZE_LDFLAGS", linker_optimize_flags.ldflags
 )
 
-add_old_configure_assignment("CPPFLAGS", os_cppflags)
-
 add_old_configure_assignment("_COMPILATION_CFLAGS", compilation_flags.cflags)
 add_old_configure_assignment("_COMPILATION_CXXFLAGS", compilation_flags.cxxflags)
 
@@ -639,6 +630,5 @@ set_config("HOST_CPPFLAGS", host_cppflags)
 set_config("HOST_CFLAGS", host_cflags)
 set_config("HOST_CXXFLAGS", host_cflags)
 set_config("HOST_LDFLAGS", host_ldflags)
-set_config("OS_CPPFLAGS", os_cppflags)
 set_config("ASFLAGS", os_asflags)
 set_config("DSO_LDOPTS", dso_flags.ldopts)
diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 6c8b4e5..4daa638 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -38,14 +38,6 @@ option(
 )
 
 
-option(
-    env="CPPFLAGS",
-    help="Extra flags for preprocessing sources.",
-    nargs=1,
-    default="",
-)
-
-
 option(
     env="ASFLAGS",
     help="Extra flags for assembling sources.",
diff --git a/js/src/old-configure.in b/js/src/old-configure.in
index 70d94fe..5f0840f 100644
--- a/js/src/old-configure.in
+++ b/js/src/old-configure.in
@@ -19,6 +19,7 @@ dnl = CFLAGS="-foo" ./configure
 dnl =
 dnl ========================================================
 CFLAGS="${CFLAGS=}"
+CPPFLAGS="${CPPFLAGS=}"
 CXXFLAGS="${CXXFLAGS=}"
 LDFLAGS="${LDFLAGS=}"
 
@@ -354,9 +355,11 @@ LDFLAGS=`echo \
 
 OS_CFLAGS="$CFLAGS"
 OS_CXXFLAGS="$CXXFLAGS"
+OS_CPPFLAGS="$CPPFLAGS"
 OS_LDFLAGS="$LDFLAGS"
 AC_SUBST_LIST(OS_CFLAGS)
 AC_SUBST_LIST(OS_CXXFLAGS)
+AC_SUBST_LIST(OS_CPPFLAGS)
 AC_SUBST_LIST(OS_LDFLAGS)
 
 AC_SUBST(HOST_CC)
diff --git a/old-configure.in b/old-configure.in
index b2f9ac8..e18c406 100644
--- a/old-configure.in
+++ b/old-configure.in
@@ -19,6 +19,7 @@ dnl = CFLAGS="-foo" ./configure
 dnl =
 dnl ========================================================
 CFLAGS="${CFLAGS=}"
+CPPFLAGS="${CPPFLAGS=}"
 CXXFLAGS="${CXXFLAGS=}"
 LDFLAGS="${LDFLAGS=}"
 
@@ -507,9 +508,11 @@ LDFLAGS=`echo \
 
 OS_CFLAGS="$CFLAGS"
 OS_CXXFLAGS="$CXXFLAGS"
+OS_CPPFLAGS="$CPPFLAGS"
 OS_LDFLAGS="$LDFLAGS"
 AC_SUBST_LIST(OS_CFLAGS)
 AC_SUBST_LIST(OS_CXXFLAGS)
+AC_SUBST_LIST(OS_CPPFLAGS)
 AC_SUBST_LIST(OS_LDFLAGS)
 
 AC_SUBST(HOST_CC)
