diff --git a/media/libyuv/libyuv/libyuv.gyp b/media/libyuv/libyuv/libyuv.gyp
index 2e5c8af..45df0d2 100644
--- a/media/libyuv/libyuv/libyuv.gyp
+++ b/media/libyuv/libyuv/libyuv.gyp
@@ -71,7 +71,9 @@
             # '-mthumb',  # arm32 not thumb
           ],
           'cflags_mozilla!': [
-            '<@(moz_neon_cflags_block_list)',
+            '-mfpu=vfp',
+            '-mfpu=vfpv3',
+            '-mfpu=vfpv3-d16',
           ],
           'conditions': [
             # Disable LTO in libyuv_neon target due to gcc 4.9 compiler bug.
@@ -160,95 +162,12 @@
               '-Wl,--dynamic-linker,/system/bin/linker',
             ],
           }],
-          ['target_arch == "armv7" or target_arch == "arm64" and moz_have_arm_i8mm_and_dot_prod == 1 and build_with_mozilla == 1', {
-            'dependencies': [
-                 ':libyuv_neon',
-            ],
-          }],
-          ['target_arch == "arm64" and moz_have_arm_sve2 == 1 and build_with_mozilla == 1', {
-            'dependencies': [
-                 ':libyuv_sve',
-            ],
-            'defines' :[
-              'LIBYUV_SVE',
-            ]
-          }],
-          ['target_arch == "arm64" and moz_have_arm_sve2 == 1 and build_with_mozilla == 1', {
-            'dependencies': [
-                 ':libyuv_sve',
-            ],
-            'defines' :[
-              'LIBYUV_SVE',
-            ]
-          }],
         ], #conditions
       },
       'sources': [
         '<@(libyuv_sources)',
       ],
     },
-    {
-      'target_name': 'libyuv_neon',
-      'type': 'static_library',
-      'variables': {
-        'optimize': 'max',  # enable O2 and ltcg.
-      },
-      'conditions': [
-        ['target_arch == "arm64" and moz_have_arm_i8mm_and_dot_prod == 1 and build_with_mozilla == 1', {
-          'cflags_mozilla': [
-            '-march=armv8.2-a+dotprod+i8mm',
-          ],
-        }],
-        ['build_neon != 0', {
-          'cflags_mozilla!': [
-            '<@(moz_neon_cflags_block_list)',
-          ],
-          'sources': [
-            '<@(libyuv_neon_sources)',
-          ],
-        }],
-     ], #conditions
-      'include_dirs': [
-        'include',
-        '.',
-      ],
-      'direct_dependent_settings': {
-        'include_dirs': [
-          'include',
-          '.',
-        ], #conditions
-      },
-    },
-    {
-      'target_name': 'libyuv_sve',
-      'type': 'static_library',
-      'variables': {
-        'optimize': 'max',  # enable O2 and ltcg.
-      },
-      'conditions': [
-        ['target_arch == "arm64" and moz_have_arm_sve2 == 1 and build_with_mozilla == 1', {
-          'cflags_mozilla!': [
-            '<@(moz_neon_cflags_block_list)',
-          ],
-          'cflags_mozilla': [
-            '-march=armv9-a+dotprod+sve2+i8mm',
-          ],
-          'sources': [
-            '<@(libyuv_sve_sources)',
-          ],
-        }],
-     ], #conditions
-      'include_dirs': [
-        'include',
-        '.',
-      ],
-      'direct_dependent_settings': {
-        'include_dirs': [
-          'include',
-          '.',
-        ], #conditions
-      },
-    },
   ], # targets.
 }
 
diff --git a/media/libyuv/libyuv/libyuv.gypi b/media/libyuv/libyuv/libyuv.gypi
index 1fd1be7..74fa0fe 100644
--- a/media/libyuv/libyuv/libyuv.gypi
+++ b/media/libyuv/libyuv/libyuv.gypi
@@ -8,29 +8,6 @@
 
 {
   'variables': {
-    'moz_have_arm_sve2%': '<(moz_have_arm_sve2)',
-    'moz_have_arm_i8mm_and_dot_prod%': '<(moz_have_arm_i8mm_and_dot_prod)',
-    'moz_neon_cflags_block_list': [
-            '-mfpu=vfp',
-            '-mfpu=vfpv3',
-            '-mfpu=vfpv3-d16',
-    ],
-    # Needs to be reflected in upstream gyp file.
-    'libyuv_sve_sources': [
-      # sources.
-      'source/row_sve.cc',
-    ],
-    'libyuv_neon_sources': [
-      # ARM Source Files
-      "source/compare_neon.cc",
-      "source/compare_neon64.cc",
-      "source/rotate_neon.cc",
-      "source/rotate_neon64.cc",
-      "source/row_neon.cc",
-      "source/row_neon64.cc",
-      "source/scale_neon.cc",
-      "source/scale_neon64.cc",
-    ],
     'libyuv_sources': [
       # includes.
       'include/libyuv.h',
@@ -63,6 +40,8 @@
       'source/compare_common.cc',
       'source/compare_gcc.cc',
       'source/compare_msa.cc',
+      'source/compare_neon.cc',
+      'source/compare_neon64.cc',
       'source/compare_win.cc',
       'source/convert.cc',
       'source/convert_argb.cc',
@@ -81,11 +60,15 @@
       'source/rotate_common.cc',
       'source/rotate_gcc.cc',
       'source/rotate_msa.cc',
+      'source/rotate_neon.cc',
+      'source/rotate_neon64.cc',
       'source/rotate_win.cc',
       'source/row_any.cc',
       'source/row_common.cc',
       'source/row_gcc.cc',
       'source/row_msa.cc',
+      'source/row_neon.cc',
+      'source/row_neon64.cc',
       'source/row_win.cc',
       'source/scale.cc',
       'source/scale_any.cc',
@@ -93,6 +76,8 @@
       'source/scale_common.cc',
       'source/scale_gcc.cc',
       'source/scale_msa.cc',
+      'source/scale_neon.cc',
+      'source/scale_neon64.cc',
       'source/scale_rgb.cc',
       'source/scale_uv.cc',
       'source/scale_win.cc',
