diff --git a/toolkit/system/gnome/nsGIOService.cpp b/toolkit/system/gnome/nsGIOService.cpp
--- a/toolkit/system/gnome/nsGIOService.cpp
+++ b/toolkit/system/gnome/nsGIOService.cpp
@@ -713,10 +713,12 @@
 NS_IMETHODIMP
 nsGIOService::GetAppForURIScheme(const nsACString& aURIScheme,
                                  nsIHandlerApp** aApp) {
   *aApp = nullptr;
 
+  // Portals works with DBus only.
+#ifdef MOZ_ENABLE_DBUS
   // Application in flatpak sandbox does not have access to the list
   // of installed applications on the system. We use SchemeSupported
   // method to check if the URI scheme is supported and then use
   // generic nsFlatpakHandlerApp which forwards launch call to the system.
   if (widget::ShouldUsePortal(widget::PortalKind::MimeHandler)) {
@@ -775,10 +777,11 @@
     LOG("Scheme '%s' is supported.\n", PromiseFlatCString(aURIScheme).get());
     RefPtr<nsFlatpakHandlerApp> mozApp = new nsFlatpakHandlerApp();
     mozApp.forget(aApp);
     return NS_OK;
   }
+#endif
 
   RefPtr<GAppInfo> app_info = dont_AddRef(g_app_info_get_default_for_uri_scheme(
       PromiseFlatCString(aURIScheme).get()));
   if (!app_info) {
     return NS_ERROR_FAILURE;


