diff --git a/build/moz.configure/finalize-flags.configure b/build/moz.configure/finalize-flags.configure
index b33bcf3..d701713 100644
--- a/build/moz.configure/finalize-flags.configure
+++ b/build/moz.configure/finalize-flags.configure
@@ -13,32 +13,6 @@
     android_flags,
     thumb_option,
 )
-@imports(_from="mozbuild.shellutil", _import="split")
-def os_ldflags(
-    env_ldflags,
-    linker_flags,
-    linker_ldflags,
-    pack_relative_relocs_flags,
-    android_flags,
-    thumb_option,
-):
-    flags = []
-    if android_flags:
-        flags.extend(android_flags.ldflags)
-    if thumb_option:
-        flags.extend(thumb_option)
-    flags.extend(split(env_ldflags[0]))
-    flags.extend(linker_flags.ldflags)
-    if linker_ldflags:
-        flags.extend(linker_ldflags)
-    if pack_relative_relocs_flags:
-        flags.extend(pack_relative_relocs_flags)
-    return flags
-
-
-add_old_configure_assignment("LDFLAGS", os_ldflags)
-set_config("OS_LDFLAGS", os_ldflags)
-
 
 @depends(linker_optimize_flags, moz_optimize_flags, lto, c_compiler)
 def moz_optimize_ldflags(linker_optimize_flags, moz_optimize_flags, lto, c_compiler):
diff --git a/build/moz.configure/flags.configure b/build/moz.configure/flags.configure
index cb696fb..9f50bfc 100644
--- a/build/moz.configure/flags.configure
+++ b/build/moz.configure/flags.configure
@@ -686,11 +686,23 @@ def add_sse2_flag_if_needed(flags):
 add_sse2_flag_if_needed(os_cflags)
 add_sse2_flag_if_needed(os_cxxflags)
 
+@depends("LDFLAGS", linker_flags, android_flags, thumb_option)
+@imports(_from="mozbuild.shellutil", _import="split")
+def os_ldflags(env_ldflags, linker_flags, android_flags, thumb_option):
+    flags = []
+    if android_flags:
+        flags.extend(android_flags.ldflags)
+    if thumb_option:
+        flags.extend(thumb_option)
+    flags.extend(split(env_ldflags[0]))
+    flags.extend(linker_flags.ldflags)
+    return flags
 
 # Please keep these last in this file.
 add_old_configure_assignment("CPPFLAGS", os_cppflags)
 add_old_configure_assignment("CFLAGS", os_cflags)
 add_old_configure_assignment("CXXFLAGS", os_cxxflags)
+add_old_configure_assignment("LDFLAGS", os_ldflags)
 
 set_config("HOST_CPPFLAGS", host_cppflags)
 set_config("HOST_CFLAGS", host_cflags)
diff --git a/js/src/old-configure.in b/js/src/old-configure.in
index dbf616d..1333a97 100644
--- a/js/src/old-configure.in
+++ b/js/src/old-configure.in
@@ -11,6 +11,10 @@ AC_INIT(js/src/jsapi.h)
 AC_CONFIG_AUX_DIR(${srcdir}/build/autoconf)
 AC_CANONICAL_SYSTEM
 
+dnl Set the minimum version of toolkit libs used by mozilla
+dnl ========================================================
+LDFLAGS="$LDFLAGS $LINKER_LDFLAGS $PACK_REL_RELOC_FLAGS"
+
 if test -z "$JS_STANDALONE"; then
   autoconfmk=autoconf-js.mk
 fi
@@ -138,8 +142,10 @@ AC_SUBST(MOZ_APP_DISPLAYNAME)
 
 OS_CFLAGS="$CFLAGS"
 OS_CXXFLAGS="$CXXFLAGS"
+OS_LDFLAGS="$LDFLAGS"
 AC_SUBST_LIST(OS_CFLAGS)
 AC_SUBST_LIST(OS_CXXFLAGS)
+AC_SUBST_LIST(OS_LDFLAGS)
 
 AC_SUBST(HOST_CC)
 AC_SUBST(HOST_CXX)
diff --git a/moz.configure b/moz.configure
index 3180ece..925358e 100755
--- a/moz.configure
+++ b/moz.configure
@@ -432,12 +432,8 @@ fold_libs = dependable(False)
 # And dmd
 dmd = dependable(False)
 
-# Only available when toolkit/moz.configure is included
-pack_relative_relocs_flags = dependable(False)
-
 include(include_project_configure)
 
-
 # Final flags validation and gathering
 # -------------------------------------------------
 
diff --git a/old-configure.in b/old-configure.in
index 1a8c809..6afcb26 100644
--- a/old-configure.in
+++ b/old-configure.in
@@ -11,6 +11,12 @@ AC_INIT(config/config.mk)
 AC_CONFIG_AUX_DIR(${srcdir}/build/autoconf)
 AC_CANONICAL_SYSTEM
 
+dnl Set the minimum version of toolkit libs used by mozilla
+dnl ========================================================
+
+
+LDFLAGS="$LDFLAGS $LINKER_LDFLAGS $PACK_REL_RELOC_FLAGS"
+
 dnl ========================================================
 dnl Checks for compilers.
 dnl ========================================================
@@ -269,8 +275,10 @@ AC_SUBST(MOZ_APP_MAXVERSION)
 
 OS_CFLAGS="$CFLAGS"
 OS_CXXFLAGS="$CXXFLAGS"
+OS_LDFLAGS="$LDFLAGS"
 AC_SUBST_LIST(OS_CFLAGS)
 AC_SUBST_LIST(OS_CXXFLAGS)
+AC_SUBST_LIST(OS_LDFLAGS)
 
 AC_SUBST(HOST_CC)
 AC_SUBST(HOST_CXX)
diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 691f444..dfef4ee 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1813,6 +1813,8 @@ with only_when("--enable-compile-environment"):
         if may_use_pack_relative_relocs:
             return flags
 
+    add_old_configure_assignment("PACK_REL_RELOC_FLAGS", pack_relative_relocs_flags)
+
     @depends(
         select_linker,
         pack_relative_relocs_flags,
