diff --git a/widget/gtk/WakeLockListener.cpp b/widget/gtk/WakeLockListener.cpp
--- a/widget/gtk/WakeLockListener.cpp
+++ b/widget/gtk/WakeLockListener.cpp
@@ -818,14 +818,16 @@
 }
 
 void WakeLockTopic::Shutdown() {
   WAKE_LOCK_LOG("WakeLockTopic::Shutdown() state %s",
                 GetInhibitStateName(mState));
+#if defined(MOZ_ENABLE_DBUS)
   if (mCancellable) {
     g_cancellable_cancel(mCancellable);
     mCancellable = nullptr;
   }
+#endif
 }
 
 bool WakeLockTopic::IsWakeLockTypeAvailable(int aWakeLockType) {
   switch (aWakeLockType) {
 #if defined(MOZ_ENABLE_DBUS)


