diff --git a/intl/components/src/TimeZone.h b/intl/components/src/TimeZone.h
--- a/intl/components/src/TimeZone.h
+++ b/intl/components/src/TimeZone.h
@@ -224,10 +224,11 @@
    * Return an enumeration over all available time zones.
    */
   static Result<SpanEnumeration<char>, ICUError> GetAvailableTimeZones();
 
  private:
+#if MOZ_INTL_USE_ICU_CPP_TIMEZONE
   template <typename B>
   static ICUResult FillBuffer(const icu::UnicodeString& aString, B& aBuffer) {
     int32_t length = aString.length();
     if (!aBuffer.reserve(AssertedCast<size_t>(length))) {
       return Err(ICUError::OutOfMemory);
@@ -243,11 +244,10 @@
     aBuffer.written(written);
 
     return Ok{};
   }
 
-#if MOZ_INTL_USE_ICU_CPP_TIMEZONE
   UniquePtr<icu::TimeZone> mTimeZone = nullptr;
 #else
   UCalendar* mCalendar = nullptr;
 #endif
 };

