diff --git a/build/moz.configure/arm.configure b/build/moz.configure/arm.configure
index b1820e3..29fc364 100644
--- a/build/moz.configure/arm.configure
+++ b/build/moz.configure/arm.configure
@@ -159,18 +159,18 @@ check_and_add_flag(
     float_abi_option,
     soft_float_option,
 )
-def all_arm_flags(arch, thumb, interwork, fpu, float_abi, soft_float):
+def all_flags(arch, thumb, interwork, fpu, float_abi, soft_float):
     return arch + thumb + interwork + fpu + float_abi + soft_float
 
 
-add_old_configure_assignment("_ARM_FLAGS", all_arm_flags)
+add_old_configure_assignment("_ARM_FLAGS", all_flags)
 add_old_configure_assignment("_THUMB_FLAGS", thumb_option)
 
 
-@depends(configure_cache, c_compiler, all_arm_flags)
+@depends(configure_cache, c_compiler, all_flags)
 @checking("ARM version support in compiler", lambda x: x.arm_arch)
 @imports(_from="textwrap", _import="dedent")
-def arm_target(configure_cache, compiler, all_arm_flags):
+def arm_target(configure_cache, compiler, all_flags):
     # We're going to preprocess the following source to figure out some details
     # about the arm target options we have enabled.
     source = dedent(
@@ -210,7 +210,7 @@ def arm_target(configure_cache, compiler, all_arm_flags):
         [compiler.compiler] + compiler.flags,
         compiler.language,
         source,
-        ["-E"] + all_arm_flags,
+        ["-E"] + all_flags,
         wrapper=compiler.wrapper,
     )
     # Metadata emitted by preprocessors such as GCC with LANG=ja_JP.utf-8 may
diff --git a/build/moz.configure/bindgen.configure b/build/moz.configure/bindgen.configure
index 8339191..2014596 100644
--- a/build/moz.configure/bindgen.configure
+++ b/build/moz.configure/bindgen.configure
@@ -326,11 +326,8 @@ set_config("MOZ_CLANG_PATH", bindgen_config_paths.clang_path)
     cxx_compiler,
     bindgen_cflags_android,
     bindgen_config_paths.clang_flags,
-    all_arm_flags,
 )
-def basic_bindgen_cflags(
-    target, is_unix, compiler_info, android_cflags, clang_flags, all_arm_flags
-):
+def basic_bindgen_cflags(target, is_unix, compiler_info, android_cflags, clang_flags):
     args = [
         "-x",
         "c++",
@@ -375,7 +372,7 @@ def basic_bindgen_cflags(
             "-DHAVE_VISIBILITY_HIDDEN_ATTRIBUTE=1",
         ]
 
-    return args + (clang_flags or []) + (all_arm_flags or [])
+    return args + (clang_flags or [])
 
 
 option(
