diff --git a/gfx/2d/SwizzleNEON.h b/gfx/2d/SwizzleNEON.h
--- a/gfx/2d/SwizzleNEON.h
+++ b/gfx/2d/SwizzleNEON.h
@@ -40,7 +40,7 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
     // Load single pixel
     dst32 = vld1q_lane_u32(src32, vdupq_n_u32(0), 0);
   }
-  return vreinterpretq_u16_u32(dst32);
+  return vreinterpretq_u8_u32(dst32);
 }
 
 template <class Arch>
@@ -48,7 +48,7 @@ template <class Arch>
 static MOZ_ALWAYS_INLINE void StoreRemainder_SIMD(
     uint8_t* aDst, size_t aLength, const xsimd::batch<uint8_t, Arch>& aSrc) {
   uint32_t* dst32 = reinterpret_cast<uint32_t*>(aDst);
-  uint32x4_t src32 = vreinterpretq_u32_u16(aSrc);
+  uint32x4_t src32 = vreinterpretq_u32_u8(aSrc);
   if (aLength >= 2) {
     // Store first 2 pixels
     vst1_u32(dst32, vget_low_u32(src32));
@@ -82,14 +82,17 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
     // Swap R and B, then add to G and A (forced to 255):
     // (((src>>16) | (src << 16)) & 0x00FF00FF) |
     //   ((src | 0xFF000000) & ~0x00FF00FF)
-    return vbslq_u16(
-        vdupq_n_u16(0x00FF), vrev32q_u16(aSrc),
+    uint16x8_t src16 = vreinterpretq_u16_u8(aSrc);
+    return vreinterpretq_u8_u16(vbslq_u16(
+        vdupq_n_u16(0x00FF), vrev32q_u16(src16),
         aOpaqueAlpha
-            ? vorrq_u16(aSrc, vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000)))
-            : aSrc.data);
+            ? vorrq_u16(src16, vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000)))
+            : src16));
   }
   if constexpr (aOpaqueAlpha) {
-    return vorrq_u16(aSrc, vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000)));
+    return vreinterpretq_u8_u16(
+        vorrq_u16(vreinterpretq_u16_u8(aSrc),
+                  vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000))));
   }
   return aSrc;
 }
@@ -106,7 +109,7 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
 
   // Load all of the interleaved low and high portions of the reciprocals
   // and combine them a single vector as lo1 hi1 lo2 hi2 lo3 hi3 lo4 hi4
-  return vreinterpretq_u16_u32(vld1q_lane_u32(
+  return vld1q_lane_u32(
       &sUnpremultiplyTable_NEON[alphaBuf[7]],
       vld1q_lane_u32(
           &sUnpremultiplyTable_NEON[alphaBuf[5]],
@@ -115,7 +118,7 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
                                         vdupq_n_u32(0), 0),
                          1),
           2),
-      3));
+      3);
 }
 
 template <class Arch>
@@ -128,7 +131,8 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
   // Transpose the interleaved low/high portions so that we produce
   // two separate duplicated vectors for the low and high portions respectively:
   // lo1 lo1 lo2 lo2 lo3 lo3 lo4 lo4 and hi1 hi1 hi2 hi2 hi3 hi3 hi4 hi4
-  uint16x8x2_t q1234lohi = vtrnq_u16(aRecip, aRecip);
+  uint16x8_t recip16 = vreinterpretq_u16_u32(aRecip);
+  uint16x8x2_t q1234lohi = vtrnq_u16(recip16, recip16);
 
   // VQDMULH is a signed multiply that doubles (*2) the result, then takes the
   // high word.  To work around the signedness and the doubling, the low
@@ -153,8 +157,9 @@ static MOZ_ALWAYS_INLINE xsimd::batch<ui
 
   // Combine to the final pixel with ((rb | (ga << 8)) & ~0xFF000000) | (aSrc &
   // 0xFF000000), which inserts back in the original alpha value unchanged.
-  return vbslq_u16(vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000)), aSrc,
-                   vsliq_n_u16(rb, ga, 8));
+  return vreinterpretq_u8_u16(
+      vbslq_u16(vreinterpretq_u16_u32(vdupq_n_u32(0xFF000000)),
+                vreinterpretq_u16_u8(aSrc), vsliq_n_u16(rb, ga, 8)));
 }
 
 template <class Arch>

