diff --git a/tools/profiler/public/MicroGeckoProfiler.h b/tools/profiler/public/MicroGeckoProfiler.h
--- a/tools/profiler/public/MicroGeckoProfiler.h
+++ b/tools/profiler/public/MicroGeckoProfiler.h
@@ -11,12 +11,25 @@
 // Gecko, it must be possible to include the header file into any C++ codebase.
 
 #ifndef MICRO_GECKO_PROFILER
 #define MICRO_GECKO_PROFILER
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 void uprofiler_register_thread(const char* aName, void* aGuessStackTop);
 void uprofiler_unregister_thread();
+void uprofiler_simple_event_marker(const char* name, char phase, int num_args,
+                                   const char** arg_names,
+                                   const unsigned char* arg_types,
+                                   const unsigned long long* arg_values);
+
+#ifdef __cplusplus
+}
+#endif
+
 
 #ifdef __cplusplus
 struct AutoRegisterProfiler {
   AutoRegisterProfiler(const char* name, char* stacktop) {
     if (getenv("MOZ_UPROFILER_LOG_THREAD_CREATION")) {
@@ -26,11 +39,6 @@
   }
   ~AutoRegisterProfiler() { uprofiler_unregister_thread(); }
 };
 #endif  // __cplusplus
 
-void uprofiler_simple_event_marker(const char* name, char phase, int num_args,
-                                   const char** arg_names,
-                                   const unsigned char* arg_types,
-                                   const unsigned long long* arg_values);
-
 #endif  // MICRO_GECKO_PROFILER


