diff --git a/dom/webserial/PosixSerialPlatformService.cpp b/dom/webserial/PosixSerialPlatformService.cpp
--- a/dom/webserial/PosixSerialPlatformService.cpp
+++ b/dom/webserial/PosixSerialPlatformService.cpp
@@ -14,10 +14,11 @@
 
 #include "SerialLogging.h"
 #include "mozilla/AsyncPlatformPipes.h"
 #include "mozilla/Maybe.h"
 #include "mozilla/Result.h"
+#include "mozilla/ResultVariant.h"
 #include "mozilla/SyncRunnable.h"
 #include "nsDirectoryServiceDefs.h"
 #include "nsIFile.h"
 #include "nsString.h"
 #include "nsThreadUtils.h"


