diff --git a/servo/components/style/build_gecko.rs b/servo/components/style/build_gecko.rs
index 734d095..d4ead97 100644
--- a/servo/components/style/build_gecko.rs
+++ b/servo/components/style/build_gecko.rs
@@ -141,7 +141,10 @@ mod bindings {
             // them.
             let mut builder = Builder::default()
                 .rust_target(RustTarget::Stable_1_25)
-                .disable_untagged_union();
+                .disable_untagged_union()
+                .clang_arg("-target")
+                .clang_arg("armv7a-unknown-linux-gnueabihf");
+
 
             let rustfmt_path = env::var_os("RUSTFMT")
                 // This can be replaced with
