diff --git a/servo/components/style/stylesheets/viewport_rule.rs b/servo/components/style/stylesheets/viewport_rule.rs
--- a/servo/components/style/stylesheets/viewport_rule.rs
+++ b/servo/components/style/stylesheets/viewport_rule.rs
@@ -752,7 +752,7 @@
                     })
                 } else {
                     None
-                };
+                }
             };
         }
 
diff --git a/servo/components/style/values/generics/mod.rs b/servo/components/style/values/generics/mod.rs
--- a/servo/components/style/values/generics/mod.rs
+++ b/servo/components/style/values/generics/mod.rs
@@ -151,7 +151,7 @@
         // approach here.
         macro_rules! predefined {
             ($($name:expr,)+) => {
-                f(&["symbols", $($name,)+]);
+                f(&["symbols", $($name,)+])
             }
         }
         include!("../../counter_style/predefined.rs");


