diff --git a/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py b/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
index 997a296..2c3225e 100755
--- a/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
+++ b/python/mozbuild/mozbuild/test/configure/test_toolchain_configure.py
@@ -1655,6 +1655,12 @@ class RustTest(BaseConfigureTest):
                                      arm_arch=7, fpu='neon', thumb2=True)),
             'thumbv7neon-unknown-linux-gnueabihf')
 
+        self.assertEqual(
+            self.get_rust_target('armv7a-unknown-linux-gnueabihf',
+                                 arm_target=ReadOnlyNamespace(
+                                     arm_arch=7, fpu='neon', thumb2=True)),
+            'thumbv7neon-unknown-linux-gnueabihf')
+
         self.assertEqual(
             self.get_rust_target('armv7-unknown-linux-gnueabihf',
                                  version='1.32.0',
