diff -ur firefox-52.2.0esr.orig/ipc/contentproc/plugin-container.cpp firefox-52.2.0esr/ipc/contentproc/plugin-container.cpp 
--- firefox-52.2.0esr/ipc/contentproc/plugin-container.cpp 2017-09-20 23:27:21.294551452 +0000
+++ firefox-52.2.0esr/ipc/contentproc/plugin-container.cpp 2017-09-20 23:28:19.874898572 +0000
@@ -161,7 +161,7 @@
 
     XRE_SetProcessType(argv[--argc]);
 
-#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+#if defined(XP_LINUX) && defined(MOZ_GMP_SANDBOX)
     // This has to happen while we're still single-threaded, and on
     // B2G that means before the Android Binder library is
     // initialized.

